src/build.cc

```diff
@@ -135,8 +135,13 @@ void BuildStatus::BuildEdgeFinished(Edge* edge,
   if (!success) {
     string outputs;
     for (vector<Node*>::const_iterator o = edge->outputs_.begin();
-         o != edge->outputs_.end(); ++o)
-      outputs += (*o)->path() + " ";
+         o != edge->outputs_.end(); ++o) {
+      if((*o)->path().find(' ') != string::npos) {
+        outputs += "'" + (*o)->path() + "' ";
```

Comment:
Given that we print all the outnames in one line, I don't think we could prevent a selected separator from being part of the outnames, unless a more complex escaping mechanism is introduced. And I guess this is not preferred.

Currently the selected separator is an empty space " ", but it conflicts with outnames containing an empty space. In this patch, a quote is introduced to resolve such conflict.

We have some options here, but I prefer option 1 or 2 below:
1. Go with a double quote (instead of a single quote in this patch). And ignore the support for outnames containing double quote chars, which seems rare. Note: a double quote can not be part of a directory name or file name on Win7 (not tested on other versions of Windows).

2. Make ninja print such information to a given file and in a parsable format like json.

3. Introduce a more complex escaping mechanism when printing in one line to the stdout.

Nico, Scott, what do you think?